### ISSUE TITLE
Gb studio for android?

### ISSUE THREAD

**Issue by flow1986** [chrismaltby/gb-studio#1039]
Hi,
as told in the Headline.

Will it be possible to have GB studio for android (phone)?

It would be nice to develop gameboygames on the go.

Greetings Flo

**Comment by BowersIndustry** [comment]
It could "technically" be possible. But the problem is that if they focus on that then the engine will not be worked on as much. So you could bring a laptop maybe?

**Comment by Anima-SodoDev** [comment]
Also, and this was said multiple times in the server, it would be very impractical.

**Comment by patrickmollohan** [comment]
I ported GB Studio to the PinePhone (a non-Android, non-Apple, open sourced Linux-based phone) and the Raspberry Pi. It works, and works well in my experience, but that is because of the convergence feature of the PinePhone (where I can hook the phone up to a TV, keyboard, and mouse and use it as a desktop environment). On a phone (without convergence), it would be very impractical indeed, as mentioned above, even with rewrites to the UI, due to how much of the screen even a single scene would take up, let alone the rest of the UI and the virtual keyboard and such (imagine trying to place an actor in a very precise location in a scene on a screen as small as the average phone screen). A tablet may give you more room to work with, but would still be really small and would not make for an ideal environment to develop games on. If your phone has some convergence feature (I think some do, but never looked into what that means for Android), it might work like it does for my PinePhone, but you would still need the TV/monitor, keyboard, and mouse. Additionally, I think last time I checked on bringing GB Studio to Android/iOS, there were various issues getting Electron to play nicely with the touch screen, or with the operating systems in general. Don't remember what issues there were specifically, but it's not as trivial as what it was for the PinePhone/Raspberry Pi.

**Comment by Spz2022** [comment]
Test in Termux?

**Comment by essayharper** [comment]
it's not impossible, And since the project is open source, there's nothing stopping somebody who knows what they're doing from doing it, just don't hold your breath.

**Comment by essayharper** [comment]
GB Studio is an open source **electron** application, with modification it could be built with [Apache Cordova](https://cordova.apache.org/) which does support android as an export method, and electron as a platform, I'll give it a go and let you know if I can accomplish anything.

**Comment by essayharper** [comment]
so yeah, as someone previously stated in this thread, electron does not work well with the touch screen, The application would need to have an entire UI built from the ground up for it.

**Comment by MesaBlack** [comment]
In my honest opinion, it's kinda like iMovie for iOS in some respects.

**Comment by essayharper** [comment]
As a matter of fact, I managed to build it not for stock android, but for wear os, in a virtual machine, I don't know why that worked but I still need to troubleshoot stock android.

**Comment by essayharper** [comment]
I might end up releasing the wear os version as a joke 👀

**Comment by maxoakland** [comment]
> In my honest opinion, it's kinda like iMovie for iOS in some respects.

What do you mean by that? iMovie for iOS is very, very stripped down compared to iMovie for Mac so that's how I'm interpreting it and I think that would probably be an issue with GBStudio for mobile

**Comment by TheRektafire** [comment]
> Test in Termux?

Yeah right now the best option just seems to be to build a c/c++ based GB assembler in termux and use that, not sure how easy that would be though since afaik the only proper GB assembler is rgbasm and thats part of rgbds so youd probably either have to try and build the entire rgbds toolchain or modify the build scripts to only build rgbasm, though building all of rgbds might not necessarily be an issue since I think most of the tools are command line tools so in theory they should all work on termux if you build them directly on the phone with termux clang, but once again i havent actually tried it myself yet so im not sure if it would work, other than that though it should be fairly easy to make games for 8 bit platforms in general on android since its not like they require particularly complex tools or file formats, worse comes to worse you could just write your music, maps, etc directly into assembly files and compile them at the same time as the game code, or you could make them in godot and export them with an editor plugin, thats one of the nice things about having a full game engine editor on android these days :)

**Comment by uh612** [comment]
Android port pls
